devito/interfaces.py

```diff
@@ -1,29 +1,102 @@
 import numpy as np
 from sympy import IndexedBase
 from tools import aligned
-
+import os
+from signal import *
+import sys
 
 __all__ = ['DenseData', 'TimeData', 'PointData']
 
 
 class DenseData(IndexedBase):
-    def __init__(self, name, shape, dtype):
+
+    def __init__(self, name, shape, dtype, disk_path=None):
         self.name = name
         self.dtype = dtype
         self.pointer = None
         self.initializer = None
+        self.disk_path = disk_path
         super(DenseData, self).__init__(name)
 
     def __new__(cls, *args, **kwargs):
         return IndexedBase.__new__(cls, args[0], shape=args[1])
 
+    # this is the default directory where memmap files will be created
+    _default_disk_path = None
+    # holds the string name of all memmap file created
+    _memmap_file_list = []
+    # default exit code used on interrupt and kill
+    _default_exit_code = 0
+
+    # functions for managing memmap files
+    # call this method to specify where you want memmap file to be created
+    @staticmethod
+    def set_default_disk_path(disk_path):
+        """call this method to specify default directory to create memmap file"""
+        if not os.path.exists(disk_path):
+            os.makedirs(disk_path)
+        DenseData._default_disk_path = disk_path
+        print("default disk path set to: " + DenseData._default_disk_path)
+
+    @staticmethod
+    def get_memmap_file_list():
+        """returns the list of string path to all memmap file created"""
+        return DenseData._memmap_file_list
+
+    # used for clean memmap file on normal exit
+    @staticmethod
+    def remove_memmap_file():
+        """call this method at the end of script to clean up generated memmpa file"""
+        for f in DenseData._memmap_file_list:
+            try:
+                os.remove(f)
+            except OSError:
+                print("error removing " + f + " skipping")
+                pass
+
+    # used for clean memmap file on signal, internal method
+    @staticmethod
+    def _remove_memmap_file_on_signal(*args):
+        DenseData.remove_memmap_file()
+        sys.exit(DenseData._default_exit_code)
+
+    # to register clean up method for chosen signals
+    @staticmethod
+    def register_remove_memmap_file_signal():
```

Comment:
Do you think it will be better to use delegation instead of inheritance, so we package the settings for memmap into a class, and any class that want to use it can access it, not just limited to DenseData and its child classes